### DISCUSSION TITLE
[Bug]  json When the character length is an even number, the auto-completion function is abnormal,

### DISCUSSION THREAD

**Discussion by Edric-Li** [microsoft/monaco-editor#3713] [Q&A]
### Reproducible in vscode.dev or in VS Code Desktop?

- [X] Not reproducible in [vscode.dev](https://vscode.dev) or VS Code Desktop

### Reproducible in the monaco editor playground?

- [ ] Not reproducible in [the monaco editor playground](https://microsoft.github.io/monaco-editor/playground.html)

### Monaco Editor Playground Code

```typescript
monaco.editor.create(document.getElementById('container'), {
	value: "{}",
	language: 'json'
});
```

### Reproduction Steps

Write json key.

### Actual (Problematic) Behavior

When the length of the key is odd, there is auto-completion, and when it is even, there is no auto-completion.

### Expected Behavior

The autocomplete function is normal.

**Comment by Edric-Li** [comment]
![scan-code](https://user-images.githubusercontent.com/50992790/197335346-44fab28c-4a0d-4532-b853-e50ae9e9646c.gif)

**Comment by tamayika** [comment]
character length has nothing to do with the behavior, but whether the suggestion is opened or closed does.

**Reply by jiang-zhong-xi** [reply]
so how to fix that?
